### PR TITLE
Password change

### PR DESCRIPTION
**Pull request by LukaszPiskadlo**
Allow existing users to change their password.

### CODE REVIEW

File: src/main/java/com/intive/patronage/toz/passwords/PasswordsController.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            @ApiResponse(code = 201, message = "Password changed"),
+            @ApiResponse(code = 400, message = "Wrong password", response = ErrorResponse.class)
+    })
+    @PreAuthorize("hasAnyAuthority('TOZ', 'VOLUNTEER')")
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.CREATED)
+    public PasswordResponseView changePassword(@Valid @RequestBody PasswordChangeRequestView passwordChangeRequest,
+                                               @ApiIgnore @AuthenticationPrincipal UserContext userContext) {
+
+        passwordsService.changePassword(userContext.getEmail(),
```
Comment (ppiotr191) [inline src/main/java/com/intive/patronage/toz/passwords/PasswordsController.java]: I think, passwordService should have only email and newPassword. If you want use oldPassword, name f.e "changePasswordForExistingUser" is more suit.

File: src/main/java/com/intive/patronage/toz/passwords/PasswordsService.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+            throw new WrongPasswordException(message);
+        }
+
+        if (passwordEncoder.matches(newPassword, user.getPasswordHash())) {
+            final String message =
+                    messageSource.getMessage("samePasswords", null, LocaleContextHolder.getLocale());
+
+            throw new WrongPasswordException(message);
+        }
```
Comment (Jakub-Izbicki) [inline src/main/java/com/intive/patronage/toz/passwords/PasswordsService.java]: Maybe you could move validation to a sepatare method? Then `changePassword()` method would be more clear.

File: src/main/java/com/intive/patronage/toz/passwords/PasswordsService.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (Jakub-Izbicki) [inline src/main/java/com/intive/patronage/toz/passwords/PasswordsService.java]: Maybe you could move validation to a sepatare method? Then `changePassword()` method would be more clear.

File: src/main/java/com/intive/patronage/toz/passwords/model/PasswordChangeRequestView.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    @ApiModelProperty(value = "Old password", required = true, position = 1)
+    @NotEmpty
+    private String oldPassword;
+
+    @ApiModelProperty(value = "New password", required = true, position = 2)
+    @NotEmpty
+    private String newPassword;
+
+    public PasswordChangeRequestView(String oldPassword, String newPassword) {
```
Comment (maras93) [inline src/main/java/com/intive/patronage/toz/passwords/model/PasswordChangeRequestView.java]: Maybe this could be replaced with @AllArgsConstructor?

**Pull request merged**